app/src/main/java/com/tuf2000m/energymeter/di/NetworkModule.kt

```diff
@@ -0,0 +1,57 @@
+package com.tuf2000m.energymeter.di
+
+import com.google.firebase.auth.FirebaseAuth
+import com.tuf2000m.energymeter.BuildConfig
+import com.tuf2000m.energymeter.data.remote.ApiService
+import com.tuf2000m.energymeter.utils.Constant
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object NetworkModule {
+
+    @Singleton
+    @Provides
+    fun provideLoggingInterceptor(): HttpLoggingInterceptor {
```

Comment:
Provides an instance of the HTTP logging interceptor used for logging network requests and responses. The logging level is set to 'BODY' for detailed logging.